In [1]:
import pandas as pd

In [2]:
dataset=pd.read_csv('Placement_Data_Full_Class.csv')

In [3]:
dataset

,sl_no,gender,ssc_p,ssc_b,hsc_p,hsc_b,hsc_s,degree_p,degree_t,workex,etest_p,specialisation,mba_p,status,salary
0,1,M,67.00,Others,91.00,Others,Commerce,58.00,Sci&Tech,No,55.0,Mkt&HR,58.80,Placed,270000.0
1,2,M,79.33,Central,78.33,Others,Science,77.48,Sci&Tech,Yes,86.5,Mkt&Fin,66.28,Placed,200000.0
2,3,M,65.00,Central,68.00,Central,Arts,64.00,Comm&Mgmt,No,75.0,Mkt&Fin,57.80,Placed,250000.0
3,4,M,56.00,Central,52.00,Central,Science,52.00,Sci&Tech,No,66.0,Mkt&HR,59.43,Not Placed,NaN
4,5,M,85.80,Central,73.60,Central,Commerce,73.30,Comm&Mgmt,No,96.8,Mkt&Fin,55.50,Placed,425000.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
210,211,M,80.60,Others,82.00,Others,Commerce,77.60,Comm&Mgmt,No,91.0,Mkt&Fin,74.49,Placed,400000.0
211,212,M,58.00,Others,60.00,Others,Science,72.00,Sci&Tech,No,74.0,Mkt&Fin,53.62,Placed,275000.0
212,213,M,67.00,Others,67.00,Others,Commerce,73.00,Comm&Mgmt,Yes,59.0,Mkt&Fin,69.72,Placed,295000.0
213,214,F,74.00,Others,66.00,Others,Commerce,58.00,Comm&Mgmt,No,70.0,Mkt&HR,60.23,Placed,204000.0


In [4]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 215 entries, 0 to 214
Data columns (total 15 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   sl_no           215 non-null    int64  
 1   gender          215 non-null    object 
 2   ssc_p           215 non-null    float64
 3   ssc_b           215 non-null    object 
 4   hsc_p           215 non-null    float64
 5   hsc_b           215 non-null    object 
 6   hsc_s           215 non-null    object 
 7   degree_p        215 non-null    float64
 8   degree_t        215 non-null    object 
 9   workex          215 non-null    object 
 10  etest_p         215 non-null    float64
 11  specialisation  215 non-null    object 
 12  mba_p           215 non-null    float64
 13  status          215 non-null    object 
 14  salary          148 non-null    float64
dtypes: float64(6), int64(1), object(8)
memory usage: 25.3+ KB


In [5]:
dataset.describe()

,sl_no,ssc_p,hsc_p,degree_p,etest_p,mba_p,salary
count,215.000000,215.000000,215.000000,215.000000,215.000000,215.000000,148.000000
mean,108.000000,67.303395,66.333163,66.370186,72.100558,62.278186,288655.405405
std,62.209324,10.827205,10.897509,7.358743,13.275956,5.833385,93457.452420
min,1.000000,40.890000,37.000000,50.000000,50.000000,51.210000,200000.000000
25%,54.500000,60.600000,60.900000,61.000000,60.000000,57.945000,240000.000000
50%,108.000000,67.000000,65.000000,66.000000,71.000000,62.000000,265000.000000
75%,161.500000,75.700000,73.000000,72.000000,83.500000,66.255000,300000.000000
max,215.000000,89.400000,97.700000,91.000000,98.000000,77.890000,940000.000000


In [11]:
Quan=[]
Qual=[]

In [12]:
for columnName in dataset.columns:
    if dataset[columnName].dtype=="O":
        Qual.append(columnName)
    else:
        Quan.append(columnName)

In [19]:
Quan

['ssc_p', 'hsc_p', 'degree_p', 'etest_p', 'mba_p', 'salary']

In [32]:
descrptive=pd.DataFrame(index=["IQR",'lesser','1.5rule',"greater"],columns=Quan)
descrptive

,ssc_p,hsc_p,degree_p,etest_p,mba_p,salary
IQR,NaN,NaN,NaN,NaN,NaN,NaN
lesser,NaN,NaN,NaN,NaN,NaN,NaN
1.5rule,NaN,NaN,NaN,NaN,NaN,NaN
greater,NaN,NaN,NaN,NaN,NaN,NaN


In [33]:
for columnName in Quan:
    descrptive[columnName]["IQR"]=dataset.describe()[columnName]['75%'] - dataset.describe()[columnName]['25%']
    descrptive[columnName]["1.5rule"]=1.5*descrptive[columnName]["IQR"]
    descrptive[columnName]["Lesser"]=dataset.describe()[columnName]['25%']-descrptive[columnName]["1.5rule"]
    descrptive[columnName]["Greater"]=dataset.describe()[columnName]['75%']+descrptive[columnName]["1.5rule"]
    

C:\Users\USER\AppData\Local\Temp\ipykernel_4580\2624840510.py:2: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!
You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.
A typical example is when you are setting values in a column of a DataFrame, like:

df["col"][row_indexer] = value

Use `df.loc[row_indexer, "col"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy

  descrptive[columnName]["IQR"]=dataset.describe()[columnName]['75%'] - dataset.describe()[columnName]['25%']
C:\Users\USER\AppData\

In [34]:
descrptive

,ssc_p,hsc_p,degree_p,etest_p,mba_p,salary
IQR,15.1,12.1,11.0,23.5,8.31,60000.0
lesser,NaN,NaN,NaN,NaN,NaN,NaN
1.5rule,22.65,18.15,16.5,35.25,12.465,90000.0
greater,NaN,NaN,NaN,NaN,NaN,NaN


These columns are numeric features of candidates:

• ssc_p – Secondary school percentage
• hsc_p – Higher secondary percentage
• degree_p – Degree percentage
• etest_p – Employability test percentage
• mba_p – MBA percentage
• salary – Offered salary

Then you see rows like:

• IQR values
• 1.5×IQR lower bound (“lesser”)
• 1.5×IQR upper bound (“greater”)

That tells us this table is not raw data.
It’s a statistical summary used for outlier detection.

Now, what the IQR method means (plain language)
IQR = Q3 − Q1
It measures the middle 50% of the data.

Using IQR, we define:
• Lower bound = Q1 − 1.5 × IQR
• Upper bound = Q3 + 1.5 × IQR

Anything outside this range is considered an outlier.

What your table is telling us (feature-wise insights)

Salary
• IQR ≈ 60,000
• Lower bound ≈ 60,000
• Upper bound ≈ 90,000

Interpretation:
Most salaries cluster tightly.
ery high salaries (>90k) are outliers, likely top-tier placements.
Very low salaries (<60k) are also unusual.

Academic percentages (ssc_p, hsc_p, degree_p, mba_p)
• IQR values are relatively small (8–15 range)
• Upper bounds are not extremely high

Interpretation:
Academic scores are well-distributed and stable.
No extreme academic performers skewing the dataset.
This suggests academic performance alone may not explain very high salaries.

etest_p (Employability test)
• Highest IQR (~23.5)
• Wider spread than academics
Interpretation:
Employability test scores vary a lot.
This feature may carry stronger predictive power for placement outcomes.